### PR TITLE
Fixed detection of CoreOS.

### PR DESCRIPTION
**Pull request by asridharan**
Also, added a DC/OS specific network configuration when we detect
`networkd` is enabled.

## High-level description

What features does this change enable? What bugs does this change fix?
Please see the JIRA ticket.

## Corresponding DC/OS tickets (obligatory)

These DC/OS JIRA ticket(s) must be updated (ideally closed) in the moment this PR lands:

  - [DCOS_OSS-2003](https://jira.mesosphere.com/browse/DCOS_OSS-2003) Backport of DCOS_OSS-1790 .

## Related tickets (optional)

Other tickets related to this change:

  - [DCOS_OSS-1790](https://jira.mesosphere.com/browse/DCOS_OSS-1790)

## Checklist for all PRs

  - [ ] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [ ] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [ ] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [ ] Change log from the last version integrated (this should be a link to commits for easy verification and review): [example](https://github.com/dcos/dcos-mesos-modules/compare/f6fa27d7c40f4207ba3bb2274e2cfe79b62a395a...6660b90fbbf69a15ef46d0184e36755881d6a5ae)
  - [ ] Test Results: [link to CI job test results for component]
  - [ ] Code Coverage (if available): [link to code coverage report]
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and review process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using GitHub's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

---
This change is [Reviewable](https://reviewable.io/reviews/dcos/dcos/2265)

### CODE REVIEW

**Comment by orsenthil** [comment]
@mesosphere-mergebot label ready for review

**Comment by asridharan** [comment]
@mesosphere-mergebot bump-ee

**Comment by asridharan** [comment]
@karlkfi @branden backport of the CoreOS fix. Could you take a look and sign off?

**Comment by asridharan** [comment]
@mesosphere-mergebot bump-ee

**Comment by asridharan** [comment]
@karlkfi @branden pinging on this review again. We need this for 1.10.5.

**Comment by karlkfi** [comment]
Why isn't this a cherry-pick of https://github.com/dcos/dcos/pull/2058 ?

File: dcos_installer/action_lib.py [hunk 1]
```diff
@@ -373,18 +373,48 @@ def uninstall_dcos(config, block=False, state_json_dir=None, async_delegate=None
 def _add_prereqs_script(chain):
     inline_script = """
 #/bin/sh
-# setenforce is in this path
-PATH=$PATH:/sbin
 
-dist=$(cat /etc/os-release | sed -n 's@^ID="\(.*\)"$@\\1@p')
+# For setenforce & xfs_info
```
Comment (karlkfi) [inline dcos_installer/action_lib.py]: Is `xfs_info` being used in the old code? If not, `/sbin` should be enough...

Comment (karlkfi) [reply dcos_installer/action_lib.py]: `wfs_info` is NOT used by the old code. It was added by my changes to master: https://github.com/dcos/dcos/blob/master/dcos_installer/action_lib.py#L543

File: dcos_installer/action_lib.py [hunk 2]
```diff
@@ -373,18 +373,48 @@ def uninstall_dcos(config, block=False, state_json_dir=None, async_delegate=None
...
+    echo "CoreOS network setup complete."
+  else
+    echo "All prerequisites already installed"
+  fi
   exit 0
 fi
 
-if ([ x$dist != 'xrhel' ] && [ x$dist != 'xcentos' ]); then
-  echo "$dist is not supported. Only RHEL and CentOS are supported" >&2
+if ([ x$distro != 'xrhel' ] && [ x$distro != 'xcentos' ]); then
```
Comment (karlkfi) [inline dcos_installer/action_lib.py]: please remove the `x` prefixes and surround the string in double quotes.

**Comment by branden** [comment]
@mesosphere-mergebot bump-ee

**Pull request merged**